src/app/pages/sharing/iscsi/authorized-access/authorized-access-form/authorized-access-form.component.ts

```diff
@@ -0,0 +1,136 @@
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component } from '@angular/core';
+import { Validators } from '@angular/forms';
+import { FormBuilder } from '@ngneat/reactive-forms';
+import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
+import { TranslateService } from '@ngx-translate/core';
+import { Observable } from 'rxjs';
+import { helptextSharingIscsi } from 'app/helptext/sharing';
+import { IscsiAuthAccess } from 'app/interfaces/iscsi.interface';
+import {
+  doesNotEqualValidator,
+  matchOtherValidator,
+} from 'app/modules/entity/entity-form/validators/password-validation/password-validation';
+import { FormErrorHandlerService } from 'app/modules/ix-forms/services/form-error-handler.service';
+import IxValidatorsService from 'app/modules/ix-forms/services/ix-validators.service';
+import { WebSocketService } from 'app/services';
+import { IxSlideInService } from 'app/services/ix-slide-in.service';
+
+@UntilDestroy()
+@Component({
+  templateUrl: './authorized-access-form.component.html',
+  styleUrls: ['./authorized-access-form.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class AuthorizedAccessFormComponent {
+  get isNew(): boolean {
+    return !this.editingAccess;
+  }
+
+  get title(): string {
+    return this.isNew
+      ? this.translate.instant('Add Authorized Access')
+      : this.translate.instant('Edit Authorized Access');
+  }
+
+  form = this.formBuilder.group({
+    tag: [null as number, [Validators.required, Validators.min(0)]],
+    user: ['', Validators.required],
+    secret: ['', [
+      Validators.minLength(12),
+      Validators.maxLength(16),
+      Validators.required,
+      this.validatorService.withMessage(
+        matchOtherValidator('secret_confirm'),
+        {
+          message: this.translate.instant('Secret and confirmation should match'),
+          forProperty: 'matchOther',
+        },
+      ),
+    ]],
+    secret_confirm: ['', Validators.required],
+    peeruser: [''],
+    peersecret: ['', [
+      this.validatorService.validateOnCondition(
+        () => this.isPeerUserSet(),
+        Validators.required,
+      ),
+      Validators.minLength(12),
+      Validators.maxLength(16),
+      this.validatorService.withMessage(
+        doesNotEqualValidator('secret'),
+        {
+          message: this.translate.instant('Secret and Peer Secret can not be the same.'),
+          forProperty: 'matchesOther',
+        },
+      ),
+      this.validatorService.withMessage(
+        matchOtherValidator('peersecret_confirm'),
+        {
+          message: this.translate.instant('Secret and confirmation should match'),
+          forProperty: 'matchOther',
+        },
+      ),
+    ]],
+    peersecret_confirm: [''],
```

Comment:
I think It's better to add validation like `peersecret`
```
      this.validatorService.validateOnCondition(
        () => this.isPeerUserSet(),
        Validators.required,
      ),
```

Reply:
Not sure I understand this point.

Reply:
I think I understand. This field will be validated by `matchOtherValidator('peersecret_confirm'),` above, which in turn will be required.